providers/adfs.go

```diff
@@ -0,0 +1,191 @@
+package providers
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/requests"
+)
+
+// AdfsProvider represents an ADFS based Identity Provider
+type AdfsProvider struct {
```

Comment:
```suggestion
type ADFSProvider struct {
```

Lets capitalize the full acronym in its naming throughout